test/components/test_saveas.py

```diff
@@ -1,15 +1,17 @@
+import sys
 import unittest
 
 import fakeswi
-import sys
 
-sys.modules['swi'] = fakeswi
+try:
+    import swi
+except ImportError:
+    # Probably running on non-RISC OS system
+    swi = sys.modules['swi']
```

Comment:
It's the same behaviour as before - `import fakeswi` will create the `swi` module in the `sys.modules['swi']`, and then the `import swi` will assign `swi` to point to that module.
On non RISC OS systems, the `import swi` will fail, so we need to do that manually. Strictly I suppose we could get rid of the `import swi`, but it might be confusing to think that it's going to call SWIs when it won't.